test/zkbob/ZkBobPool.t.sol

```diff
@@ -348,6 +348,39 @@ abstract contract AbstractZkBobPoolTest is AbstractForkTest {
         vm.stopPrank();
     }
 
+    function testCancelForcedExit() public {
+        bytes memory data = _encodePermitDeposit(int256(0.5 ether / D), 0.01 ether / D);
+        _transact(data);
+
+        uint256 nullifier = _randFR();
+        pool.commitForcedExit(user2, user2, 0.4 ether / D / denominator, 128, nullifier, _randFR(), _randProof());
+        uint256 exitStart = block.timestamp + 1 hours;
+        uint256 exitEnd = block.timestamp + 24 hours;
+        bytes32 hash = pool.committedForcedExits(nullifier);
+        assertNotEq(hash, bytes32(0));
+
+        vm.expectRevert("ZkBobPool: exit not expired");
+        pool.cancelForcedExit(nullifier, user2, user2, 0.4 ether / D / denominator, exitStart, exitEnd);
+        vm.expectRevert("ZkBobPool: already exists");
+        pool.commitForcedExit(user2, user2, 0.4 ether / D / denominator, 128, nullifier, _randFR(), _randProof());
+
+        skip(12 hours);
+
+        vm.expectRevert("ZkBobPool: exit not expired");
+        pool.cancelForcedExit(nullifier, user2, user2, 0.4 ether / D / denominator, exitStart, exitEnd);
+        vm.expectRevert("ZkBobPool: already exists");
+        pool.commitForcedExit(user2, user2, 0.4 ether / D / denominator, 128, nullifier, _randFR(), _randProof());
+
+        skip(24 hours);
+
+        pool.cancelForcedExit(nullifier, user2, user2, 0.4 ether / D / denominator, exitStart, exitEnd);
```

Comment:
Should it also be tested that `executeForcedExit` is not possible after cancelation?

Reply:
Since it will be already expired by then, it will be rejected anyway at the timestamp check, so this test won't add any new coverage.

Reply:
Got you, thanks! If it is assumed that the cancelation happens in order to commit the proof one more time, does it make sense to extend the method with the proof and if the proof/index/out_commit is not empty, the force exit will be re-commited? It will allow users to save on gas fees a bit.

Reply:
I re-worked code a bit, so it's no longer that simple. But anyway I think it should be possible to make cancel and commit in the same transaction using any multicall available on-chain, since they both do not depend on `msg.sender`.